### PR TITLE
Add auto doc gen for ORTModule API during CI build

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
This PR adds tooling to auto generate Python API documentation based on sphinx

User can generate documentation locally by using `--gen-api-doc` switch on `build.sh`

ONNX Runtime Training CPU pipeline will call doc gen script to provide latest documentation

### CODE REVIEW

File: docs/python/training/conf.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#
+# Configuration file for the Sphinx documentation builder.
+
+# -- Path setup --------------------------------------------------------------
+
+import os
+import sys
+
+# ORTModule source-code folder
+sys.path.insert(0, os.path.abspath(os.path.join('..','..','..','..','orttraining','orttraining','python','training')))
```
Comment (baijumeswani) [inline docs/python/training/conf.py:14]: may we instead use relative path from the build directory to ```onnxruntime/training```?

File: docs/python/training/conf.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+print(os.path.abspath(os.path.join('..','..','..','..','orttraining','orttraining','python','training')))
```
Comment (baijumeswani) [inline docs/python/training/conf.py:15]: Is this a debug print statement? Shall we remove it or print a more specific message?

File: docs/python/training/conf.py (hunk 1 above)
Comment (thiagocrepaldi) [reply docs/python/training/conf.py:14]: PATH env var must be absolute path

File: docs/python/training/conf.py (hunk 2 above)
Comment (thiagocrepaldi) [reply docs/python/training/conf.py:15]: ooops, it is. I will remove in the next PR, in which I will integrate with Liqun work

**Review by baijumeswani — APPROVED** [review]
Looks good.

**Pull request merged**